src/widgets/Ruler.h

```diff
@@ -152,143 +144,43 @@ class AUDACITY_DLL_API Ruler {
    void Invalidate();
 
  private:
-   struct TickSizes;
-
-   class Label {
-    public:
-      double value;
-      int pos;
-      int lx, ly;
-      TranslatableString text;
-
-      void Draw(wxDC &dc, bool twoTone, wxColour c) const;
-   };
-   using Labels = std::vector<Label>;
-
-   using Bits = std::vector< bool >;
 
    void ChooseFonts( wxDC &dc ) const;
 
    void UpdateCache( wxDC &dc, const Envelope* envelope ) const;
-
-   struct Updater;
    
 public:
    bool mbTicksOnly; // true => no line the length of the ruler
    bool mbTicksAtExtremes;
 
 private:
+   RulerStruct mRulerStruct;
+
    wxColour mTickColour;
    wxPen mPen;
 
-   int          mLeft, mTop, mRight, mBottom;
-   int          mLength;
-
-   std::unique_ptr<Fonts> mpUserFonts;
-   mutable std::unique_ptr<Fonts> mpFonts;
+   std::unique_ptr<RulerStruct::Fonts> mpUserFonts;
 
-   double       mMin, mMax;
-   double       mHiddenMin, mHiddenMax;
+   std::unique_ptr<Updater> mpUpdater;
 
-   Bits mUserBits;
+   Updater::Bits mUserBits;
 
-   static std::pair< wxRect, Label > MakeTick(
-      Label lab,
-      wxDC &dc, wxFont font,
-      std::vector<bool> &bits,
-      int left, int top, int spacing, int lead,
-      bool flip, int orientation );
+   Updater::Labels mCustomMajorLabels;
+   Updater::Labels mCustomMinorLabels;
+   Updater::Labels mCustomMinorMinorLabels;
 
    struct Cache;
    mutable std::unique_ptr<Cache> mpCache;
 
    // Returns 'zero' label coordinate (for grid drawing)
-   int FindZero( const Labels &labels ) const;
+   int FindZero( const Updater::Labels &labels ) const;
 
    int GetZeroPosition() const;
 
-   int          mOrientation;
-   int          mSpacing;
-   double       mDbMirrorValue;
    bool         mHasSetSpacing;
-   bool         mLabelEdges;
-   RulerFormat  mFormat;
-   bool         mLog;
-   bool         mFlip;
-   bool         mCustom;
    bool         mbMinor;
-   TranslatableString mUnits;
    bool         mTwoTone;
    const ZoomInfo *mUseZoomInfo;
```

Comment:
Reviewing the first again, I see the `Ruler::mUseZoomInfo` becomes unused at the first commit.  Reassigned in `Ruler::SetUpdater` but not used elsewhere for anything.  So eliminate it.